src/Lykke.Service.Nem.Api/Services/NemApi.cs

```diff
@@ -0,0 +1,311 @@
+
+            var fromAddress = action.From.Split(AddressSeparator)[0];
+            var owned = await new AccountHttp(_nemUrl).MosaicsOwned(Address.CreateFromEncoded(fromAddress));
+            var required = fee.levies
+                .Append(Xem.CreateAbsolute(fee.fee))
+                .Append(mosaic)
+                .GroupBy(m => new { m.NamespaceName, m.MosaicName })
+                .Select(g => new Mosaic(g.Key.NamespaceName, g.Key.MosaicName, g.Aggregate(0UL, (v, m) => v += m.Amount)))
+                .ToList();
+
+            foreach (var req in required)
+            {
+                var own = owned.FirstOrDefault(m => m.NamespaceName == req.NamespaceName && m.MosaicName == req.MosaicName)?.Amount ?? 0UL;
+                if (own < req.Amount)
+                {
+                    throw new BlockchainException(BlockchainErrorCode.NotEnoughBalance, 
+                        $"Not enough {req.NamespaceName}:{req.MosaicName}");
+                }
+            }
+
+            var networkTime = (int)(await nodeHttp.GetExtendedNodeInfo()).NisInfo.CurrentTime;
+
+            var tx = TransferTransaction.Create(
+                networkType, 
+                new Deadline(networkTime + _expiresInMinutes * 60),
+                fee.fee,
+                Address.CreateFromEncoded(toAddress),
+                new List<Mosaic> { mosaic }, 
+                message, 
+                networkTime);
+
+            return (
+                tx.ToJson(),
+                Convert.ToDecimal(fee.fee * 1e-6),
+                tx.Deadline.GetInstant()
+            );
+        }
+
+        public bool CanGetBalances => false;
+
+        public Task DeleteAddressObservationAsync(string address) => Task.CompletedTask;
+
+        public Task<BlockchainBalance[]> GetBalancesAsync(string[] addresses, Func<string, Task<IAsset>> getAsset) =>
+            Task.FromException<BlockchainBalance[]>(new NotImplementedException());
+
+        public CapabilitiesResponse GetCapabilities()
+        {
+            return new CapabilitiesResponse()
+            {
+                AreManyInputsSupported = false,
+                AreManyOutputsSupported = false,
+                IsTransactionsRebuildingSupported = false,
+                IsTestingTransfersSupported = true,
+                IsPublicAddressExtensionRequired = true,
+                IsReceiveTransactionRequired = false,
+                CanReturnExplorerUrl = !string.IsNullOrEmpty(_explorerUrl),
+                IsAddressMappingRequired = false,
+                IsExclusiveWithdrawalsRequired = false
+            };
+        }
+
+        public ConstantsResponse GetConstants()
+        {
+            return new ConstantsResponse
+            {
+                PublicAddressExtension = new PublicAddressExtensionConstantsContract 
+                { 
+                    BaseDisplayName = "Address",
+                    DisplayName = "Message",
```

Comment:
Change to "Plain (not encrypted) message"

Reply:
Fixed